### PR TITLE
[systemtest] KafkaST: fix failed tests

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lkral@redhat.com>

### Type of change

- Bugfix

### Description

Fix failed tests in KafkaST from our job:
```
io.strimzi.systemtest.KafkaST.testKafkaJBODDeleteClaimsTrueFalse - OCP3, OCP4 - fixed bad wait for PVC deletion
io.strimzi.systemtest.KafkaST.testKafkaJBODDeleteClaimsFalse - OCP3 - race condition, tried 5 times, always green
io.strimzi.systemtest.KafkaST.testLabelModificationDoesNotBreakCluster - OCP4 - race condition, 5 times green
io.strimzi.systemtest.KafkaST.testCustomAndUpdatedValues - OCP4 - bad order of waitTillSsHasRolled
```

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Comment by im-konge** [comment]
@strimzi-ci run tests profile=regression testcase=KafkaST#testKafkaJBODDeleteClaimsTrue+testKafkaJBODDeleteClaimsFalse+testKafkaJBODDeleteClaimsTrueFalse test_only

**Comment by im-konge** [comment]
@strimzi-ci run tests profile=regression testcase=KafkaST#testKafkaJBODDeleteClaimsTrue+testKafkaJBODDeleteClaimsFalse+testKafkaJBODDeleteClaimsTrueFalse test_only

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=KafkaST#testKafkaJBODDeleteClaimsTrue+testKafkaJBODDeleteClaimsFalse+testKafkaJBODDeleteClaimsTrueFalse test_only

File: systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java [hunk 1]
```diff
@@ -1388,6 +1401,33 @@ void verifyVolumeNamesAndLabels(int kafkaReplicas, int diskCountPerReplica, int
         }
     }
 
+    void waitForPVCDeletion(int kafkaReplicas, JbodStorage jbodStorage) {
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java]: This should be in some utils class.

File: systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java [hunk 2]
```diff
@@ -1388,6 +1401,33 @@ void verifyVolumeNamesAndLabels(int kafkaReplicas, int diskCountPerReplica, int
...
+        TestUtils.waitFor("Wait for PVC deletion", Constants.POLL_INTERVAL_FOR_RESOURCE_DELETION, Constants.TIMEOUT_FOR_RESOURCE_CREATION, () -> {
+            List<String> pvcs = kubeClient().listPersistentVolumeClaims().stream()
+                    .map(pvc -> pvc.getMetadata().getName())
+                    .collect(Collectors.toList());
+            boolean isCorrect = false;
+
+            for (SingleVolumeStorage singleVolumeStorage : jbodStorage.getVolumes()) {
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java]: Wouldn't be better to create list of storages, which shouldn't be deleted after kafka deletion and compare it with `pvcs` list?

**Pull request merged**